src/main/java/com/direwolf20/buildinggadgets/common/building/placement/NoBorrow.java

```diff
@@ -0,0 +1,28 @@
+package com.direwolf20.buildinggadgets.common.building.placement;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * <p>
+ * When the annotation is applied to a method or a field, the given value, should not be copied as a reference anywhere.
+ * This means the usage of the value should be kept pure. It is not guaranteed that errors and unrecoverable errors will occur.
+ * In other words, it must not keep a direct reference the value, but making a copy is fine.
+ * </p>
+ *
+ * <p>
+ * When the annotation is applied to a parameter, it means the method will kept the usage of that parameter pure.
+ * In other words, it is guaranteed that it will not copy a direct reference to the parameter, but making a copy of the parameter is fine.
+ * </p>
+ */
+@Retention(RetentionPolicy.SOURCE)
+@Target({
+        ElementType.METHOD,
+        ElementType.FIELD,
+        ElementType.PARAMETER
+})
+public @interface NoBorrow {
```

Comment:
Please don't do a NoBorrow interface... Just make copies (I know this isn't good for Performance, but the compiler can't enforce your NoBorrow in Java...)

Reply:
??? This is an annotation, and it is source only, so it does nothing at compile-time and runtime. The point of this annotation is I don't need to copy a paragraph every time I need to mention it.

Reply:
That's exactly what I said - I just meant don't write that paragraph: do defensive copies. Optimization is nice, except when it comes to trusting other people who might not be you. And something like keeping a stray reference is a special candidate for making mistakes or actively attempting to corrupt us (though the latter is rater unlikely).

Reply:
Actually, I just realized MutableBlockPos has all of its inherited methods return a immutable copy, so this whole annotation is not even needed...